### PR TITLE
fix(ci): remove nsp

### LINKED ISSUE #596: Remove nsp
nsp is being deprecated on sep 30.

### LINKED ISSUE #597: Add `npm audit` support
Add to travis, do not fail at first.

Ref #596

### PR DESCRIPTION
**Pull request by vladikoff**
Fixes #596
Fixes #597

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -31,6 +31,7 @@ env:
   - CXX=g++-4.8 NODE_ENV=test DB=mysql
 
 before_install:
+  - npm i -g npm@6
```
Comment (pdehaan) [inline .travis.yml]: Not sure if it matters (or if you want the latest npm@6 version), but Node v8.12 includes npm 6.4.1, per https://nodejs.org/en/download/

> Latest LTS Version: 8.12.0 (includes npm 6.4.1)

**Comment by rfk** [comment]
We should also remove `grunt-nsp` from the dependencies, which is the source of all the remaining "Manual Review" audit warnings.

**Comment by vladikoff** [comment]
> We should also remove grunt-nsp from the dependencies, which is the source of all the remaining "Manual Review" audit warnings.

Removed nsp!

File: .travis.yml [hunk 2]
```diff
@@ -41,4 +42,5 @@ before_script:
 script:
   - npm run outdated
   - npm test
-  - grunt nsp --force
+  # HACK: ignore npm audit errors for now until we get them all fixed
```
[the remark this answers is not in the document]
Comment (rfk) [reply .travis.yml]: Yep; 👍 :shipit:

**Pull request merged**